config/routes.rb

```diff
@@ -4,9 +4,14 @@
 
   namespace 'api' do
     namespace 'v1', :constraints => {format: 'json'} do
-      resources :registrations, :controller => "users", :only=> [:create]
+      resources :registrations, :controller => "users", :only => [:create]
+      resources :users, :controller => "users", :only => [:update, :show, :destroy] do
+        collection { post 'check_user', :to => "users#check_user_exists"  }
```

Comment:
You don't need to map a route to a controller action like this and try to use the new hash syntax,

resources :users, only: [:update, :show, :destroy] do
  collection do 
     post :check_user
end
end, will do and it's much more cleaner